spec/index.md

```diff
@@ -5348,8 +5416,70 @@ S with
 
 ```
 
+#### Trimming canister logs
+
+Canister logs can be trimmed if the total length of their contents exceeds 4KiB.
+
+Conditions
+
+```html
+
+S.canister_logs[CanisterId] = Older_logs · Newer_logs
+SUM { |l.content| | l <- Older_logs } > 4KiB
+
+```
+
+State after
+
+```html
+
+S with
+    canister_logs[CanisterId] = Newer_logs
+
+```
+
+#### IC Management Canister: Canister logs (query call) {#ic-mgmt-canister-fetch-logs}
+
+This section specifies management canister query calls.
+They are calls to `/api/v2/canister/<effective_canister_id>/query`
+with CBOR body `Q` such that `Q.canister_id = ic_principal`.
+
+The management canister offers the method `fetch_logs`
+that can be called as a query call and
+returns logs of a requested canister.
+
+Conditions
+
+```html
+
+Q.canister_id = ic_principal
+Q.method_name = 'fetch_logs'
+Q.arg = candid(A)
+A.canister_id = effective_canister_id
+S[A.canister_id].log_visibility = Public or M.caller ∈ S.controllers[A.canister_id]
+
+```
+
+Query response `R`:
+
+```html
+
+{status: "replied"; reply: {arg: candid(S.canister_logs[A.canister_id])}, signatures: Sigs}
```

Comment:
Will we include node signatures?

Reply:
I think so since the HTTP handler attaches them for all requests to the query endpoint, but I'd like to double check.

Reply:
@ielashi PTAL

Reply:
@mraszyk Can you point to where in the HTTP handler adds signatures to the queries? And does it matter whether or not it's attached to the `fetch_canister_logs` endpoint?

Reply:
The signature is computed [here](https://github.com/dfinity/ic/blob/6c35853d3ac8330485068e0b2d8f44c2aa073ca4/rs/http_endpoints/public/src/query.rs#L252). So as long as the query execution service is used, we should be getting query signatures.